### PR TITLE
Dependencies: Upgrade `@joshwooding/vite-plugin-react-docgen-typescript`

### PR DESCRIPTION
**Pull request by brendo**
## What I did

This PR bumps the `@joshwooding/vite-plugin-react-docgen-typescript` in the `react-vite`.

This main desire behind this change is to get https://github.com/joshwooding/vite-plugin-react-docgen-typescript/pull/35 which removes a dependency on `glob-promise` and raises the minimum version of `glob` from 7 to 10.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [x] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

| name | before | after | diff | z | % |
| ---- | ------ | ----- | ---- | - | - |
| createSize |  0 B | 0 B | 0 B | - | - |
| generateSize |  76.3 MB | 76.3 MB | 331 B | -0.38 | 0% |
| initSize |  169 MB | 183 MB | 14.4 MB | **444.16** | 🔺7.9% |
| diffSize |  92.8 MB | 107 MB | 14.4 MB | **266.66** | 🔺13.5% |
| buildSize |  7.46 MB | 7.46 MB | 91 B | 1.15 | 0% |
| buildSbAddonsSize |  1.62 MB | 1.62 MB | 0 B | 1.2 | 0% |
| buildSbCommonSize |  195 kB | 195 kB | 0 B | - | 0% |
| buildSbManagerSize |  2.3 MB | 2.3 MB | 48 B | **1.52** | 0% |
| buildSbPreviewSize |  351 kB | 351 kB | 0 B | 0.99 | 0% |
| buildStaticSize |  0 B | 0 B | 0 B | - | - |
| buildPrebuildSize |  4.46 MB | 4.46 MB | 48 B | 1.16 | 0% |
| buildPreviewSize |  3 MB | 3 MB | 43 B | 1.04 | 0% |
| testBuildSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbAddonsSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbCommonSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbManagerSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbPreviewSize |  0 B | 0 B | 0 B | - | - |
| testBuildStaticSize |  0 B | 0 B | 0 B | - | - |
| testBuildPrebuildSize |  0 B | 0 B | 0 B | - | - |
| testBuildPreviewSize |  0 B | 0 B | 0 B | - | - |

| name | before | after | diff | z | % |
| ---- | ------ | ----- | ---- | - | - |
| createTime |  25.7s | 15.2s | -10s -525ms | -0.14 | -69% |
| generateTime |  19.3s | 22s | 2.7s | **1.33** | 🔺12.3% |
| initTime |  18.2s | 19.1s | 849ms | 1.09 | 4.4% |
| buildTime |  12.1s | 15.1s | 2.9s | **2.09** | 🔺19.7% |
| testBuildTime |  0ms | 0ms | 0ms | - | - |
| devPreviewResponsive |  7.1s | 7.1s | 85ms | -0.3 | 1.2% |
| devManagerResponsive |  4.6s | 4.8s | 186ms | -0.07 | 3.8% |
| devManagerHeaderVisible |  710ms | 871ms | 161ms | 0.54 | 18.5% |
| devManagerIndexVisible |  742ms | 905ms | 163ms | 0.48 | 18% |
| devStoryVisibleUncached |  1.2s | 1.2s | -29ms | -0.5 | -2.4% |
| devStoryVisible |  743ms | 906ms | 163ms | 0.49 | 18% |
| devAutodocsVisible |  625ms | 755ms | 130ms | 0.11 | 17.2% |
| devMDXVisible |  648ms | 740ms | 92ms | 0.09 | 12.4% |
| buildManagerHeaderVisible |  676ms | 735ms | 59ms | -0.14 | 8% |
| buildManagerIndexVisible |  681ms | 742ms | 61ms | -0.12 | 8.2% |
| buildStoryVisible |  753ms | 817ms | 64ms | 0.04 | 7.8% |
| buildAutodocsVisible |  697ms | 718ms | 21ms | 0.14 | 2.9% |
| buildMDXVisible |  674ms | 770ms | 96ms | **1.66** | 🔺12.5% |

### CODE REVIEW

**Review by ndelangen — APPROVED** [review]

**Comment by brendo** [comment]
👋 Do you need anything from me to push this through?

**Review by shilman — APPROVED** [review]

**Comment by shilman** [comment]
Thanks for your contribution and for your patience @brendo ! Will try to merge/release today 🙏

**Comment by shilman** [comment]
@brendo FYI, we're seeing a perf issue on our unit tests and also a dramatically increased install size. We'll need to resolve those before we can merge.

**Comment by JReinhold** [comment]
The new version of `@joshwooding/vite-plugin-react-docgen-typescript` does grow in size, purely because `glob` does. I don't understand how that would lead to a 14 MB increase though

| v0.3.1 | v0.4.0 |
|--------|--------|
| [image](https://deptree.rschristian.dev/?q=@joshwooding/vite-plugin-react-docgen-typescript@0.3.1) | [image](https://deptree.rschristian.dev/?q=@joshwooding/vite-plugin-react-docgen-typescript@0.4.0) |
| [image](https://pkg-size.dev/@joshwooding%2Fvite-plugin-react-docgen-typescript@0.3.1?no-peers) | [image](https://pkg-size.dev/@joshwooding%2Fvite-plugin-react-docgen-typescript@0.4.0?no-peers) |

**Comment by brendo** [comment]
Oh that's really quite unfortunate. From poking at `glob`, it looks like `jackspeak` is used to power the [`bin`](https://github.com/isaacs/node-glob/blob/feaf0a85c1d1bcc7c992f45edf51aaab8ed729ac/src/bin.mts#L4) and not for runtime operation. Would this mean it's tree-shaken away in typical apps?

**Comment by ndelangen** [comment]
> @ndelangen is anything stopping us from prebundling `@joshwooding/vite-plugin-react-docgen-typescript`, to circumven the transitive dependency of `jackspeak` (`@joshwooding/vite-plugin-react-docgen-typescript` -> `glob@^10` > `jackspeak`.

It's not possible for me to say if it's possible without looking at the code of `@joshwooding/vite-plugin-react-docgen-typescript`.

If the runtime spawns a subprocess that's from it's own package; then it cannot.
If the runtime spawns a subprocess that's from another package, that package has to be added as a regular dep.
If the runtime tries to load a file relative to the runtime.
if the runtime tries to resolve something relative to the runtime.

Basically:
If `@joshwooding/vite-plugin-react-docgen-typescript` makes a runtime reference to itself or another package, that isn't a normal import bundlers can deal with, then you can't prebundle it.

That would include things such as:
- `require.resolve('@joshwooding/vite-plugin-react-docgen-typescript/transformer')`
- `fs.readFile(__dirname + '/file.json')`

Those code section, the bundler would leave untouched, but then their behavior most certainly not function as intended.

**Comment by JReinhold** [comment]
@brendo if prebundling doesn't work for some reason (it throws errors at runtime), alternatively you can try to _just_ prebundle `@joshwooding/vite-plugin-react-docgen-typescript` by moving it to `devDependencies`, but _not_ prebundle `react-docgen-typescript` where the magic happens, by adding that as an explicit `dependencies`.

**Comment by JReinhold** [comment]
I'm conflicted, ease of use vs package size.
@shilman thoughts on 👆?

**Comment by xfournet** [comment]
It can be closed because #29724 has been merged
